src/unstract/adapters/llm/mistral/src/mistral.py

```diff
@@ -48,7 +46,7 @@ def get_json_schema() -> str:
 
     def get_llm_instance(self) -> Optional[LLM]:
         try:
-            llm = MistralAI(
+            llm: Optional[LLM] = MistralAI(
```

Comment:
@gaya3-zipstack instead of making all these methods return `Optional[LLM]` | `Optional[Embedding]` | `Optional[VectorDB]`, can we change the prototype to
```
def get_llm_instances(self) -> LLM:
```
and always raise an exception?
I don't think we suppress this anywhere yet

Reply:
Yes that can be done. I am not sure of how much work it is w.r.t refactoring the None checks in all the callers' context. Will try to address as much as possible.

Reply:
Yes, thank you @gaya3-zipstack . I think we can initially handle it in the SDK's context and later on in our services. Also since its only a `None` check - retaining it would do no harm as well (until we remove it).
Also please try to raise specific errors (can create new ones too) from [exceptions.py](https://github.com/Zipstack/unstract-adapters/blob/0c55bf8079c8026632cb305274d8aa854a23302a/src/unstract/adapters/exceptions.py#L14-L27)

Reply:
I will add this incrementally @chandrasekharan-zipstack on this branch.

Reply:
Thanks for handling this big change, it would definitely improve our development :)